### PR TITLE
fix upload photo WinError 32

### PR DESCRIPTION
**Pull request by trebacz626**
Fix for issue #807

### CODE REVIEW

**Review by smolyakov26 — APPROVED** [review]

**Review by smolyakov26 — CHANGES_REQUESTED** [review]
1
2019-02-24 16:29:36,399 - INFO - Logged-in successfully as 'alx_smolyakov'!
upload: First test
Analizing `./pics\1 First test.jpg`
FOUND w:834, h:720, ratio=1.1583333333333334
Horizontal image
Saving new image w:834 h:720 to `./pics\1 First test.jpg.CONVERTED.jpg`
FOUND: w:834 h:720 r:1.1583333333333334
[WinError 32] Процесс не может получить доступ к файлу, так как этот файл занят другим процессом: './pics\\1 First test.jpg.CONVERTED.jpg' -> './pics\\1 First test.jpg.CONVERTED.jpg.REMOVE_ME'
upload: First test
Analizing `./pics\1 First test.jpg`
FOUND w:834, h:720, ratio=1.1583333333333334
Horizontal image
Saving new image w:834 h:720 to `./pics\1 First test.jpg.CONVERTED.jpg`
FOUND: w:834 h:720 r:1.1583333333333334
[WinError 32] Процесс не может получить доступ к файлу, так как этот файл занят другим процессом: './pics\\1 First test.jpg.CONVERTED.jpg' -> './pics\\1 First test.jpg.CONVERTED.jpg.REMOVE_ME'

**Comment by trebacz626** [comment]
Sorry, it's my first Pull Request. I think I've messed sth with branches. Could you check if those lines have changed in your function. I'm asking because I copied this project again to see if I get any error and I got the same as you, but then I switched branch to fix-WinError32-photo-upload and everything worked properly.

**Comment by smolyakov26** [comment]
Cheked those lines , it changed, but still getting that error (

вс, 24 фев 2019 г., 18:21 trebacz626 <notifications@github.com>:

File: instabot/api/api_photo.py [hunk 1]
```diff
@@ -95,12 +95,13 @@ def upload_photo(self, photo, caption=None, upload_id=None, from_video=False):
         self.logger.info('Photo does not have a compatible '
                          'photo aspect ratio.')
         return False
+    photoFile =open(photo, 'rb')
```
Comment (ohld) [inline instabot/api/api_photo.py]: you should use this construction:
``` python
with open(photo, 'rb') as photofile:
  <your code here>
```

**Comment by smolyakov26** [comment]
which script do you use ? autopost or photo ?
and what python version do you use ?

**Comment by trebacz626** [comment]
I tested both. Python version: 3.7.2

**Review by smolyakov26 — CHANGES_REQUESTED** [review]
still getting ducking error Win32, as far as i understand it cant manage converted image.

**Review by ohld — APPROVED** [review]

**Review by smolyakov26 — CHANGES_REQUESTED** [review]
`C:\Users\фдуч\Desktop\instabot-master\examples\photos/media/Chrysanthemum.jpg.CONVERTED.jpg

may be it caused by the way to the file , it says the location that not exist on windows cant be reacheable...
MB its all abot SLASHES and BACKSLASHES ?

**Comment by lmgtrigo** [comment]
Same error...
[WinError 32] The process cannot access the file because it is being used by another process: './pics\\1 First test.jpg.CONVERTED.jpg' -> './pics\\1 First test.jpg.CONVERTED.jpg.REMOVE_ME'

**Review by ohld — APPROVED** [review]

**Pull request merged**